py/mpconfig.h

```diff
@@ -26,23 +26,28 @@
 #ifndef MICROPY_INCLUDED_PY_MPCONFIG_H
 #define MICROPY_INCLUDED_PY_MPCONFIG_H
 
-// Current version of MicroPython
+// Current version of MicroPython. This is used by sys.implementation.version
+// as well as a fallback to generate MICROPY_GIT_TAG if the git repo or tags
+// are unavailable.
 #define MICROPY_VERSION_MAJOR 1
-#define MICROPY_VERSION_MINOR 20
+#define MICROPY_VERSION_MINOR 22
 #define MICROPY_VERSION_MICRO 0
+#define MICROPY_VERSION_PRERELEASE 1
 
-// Combined version as a 32-bit number for convenience
-#define MICROPY_VERSION ( \
```

Comment:
This is needed, eg, for 3rd party libraries to detect which MicroPython version they are building against.

Reply:
Can you give an example so I can add a comment? Why would they use this instead of directly using `MICROPY_VERSION_MAJOR` etc?

Reply:
See ulab, eg `#if !defined(MICROPY_VERSION) || MICROPY_VERSION <= 70144`.

Reply:
Ah, that isn't very readable. I've made this more useful by adding a macro to generate the version.

e.g. the ulab example could now be

```c
#if MICROPY_VERSION <= MICROPY_MAKE_VERSION(1, 18, 0)
```

Reply:
They should just use a hex number, that was the intention (see how mbedtls does it, also we make an lwip version macro in a similar way).